client/extensions/woocommerce/components/d3/horizontal-bar/index.js

```diff
@@ -35,7 +35,9 @@ const HorizontalBar = ( { className, data, extent, currency, height, width } ) =
 			.attr( 'fill', 'white' )
 			.text( formatValue( data, numberFormat, currency ) );
 		const textMargin = data === 0 ? 0 : 5;
-		const isOffsetText = xPos - ( text.node().getBoundingClientRect().width + textMargin ) <= 0;
+		const isOffsetText =
+			xPos + ( text.node().getBoundingClientRect().width + textMargin * 2 ) <= scale( extent[ 1 ] );
+
```

Comment:
Something is not quite right here.

![screen shot 2018-04-03 at 1 09 36 pm](https://user-images.githubusercontent.com/1922453/38223743-82a4da38-3740-11e8-82fe-91298b35acff.png)

The previous way, for reference

![screen shot 2018-04-03 at 1 08 46 pm](https://user-images.githubusercontent.com/1922453/38223752-916b7496-3740-11e8-85fe-a6ef2d7e5ed3.png)

What's the motivation for referencing the the right edge as opposed to the left edge?

Reply:
> Something is not quite right here.

Good catch. My edit doesn't check that the text can fit inside the bar if it needs to.

> What's the motivation for referencing the the right edge as opposed to the left edge?

I think it looks and reads better.